private/tools/java/rules/jvm/external/maven/Outdated.java

```diff
@@ -44,22 +42,48 @@ public static String getReleaseVersion(String repository, String groupId, String
     }
 
     // example maven-metadata.xml
-    //<metadata>
-    //  <versioning>
-    //    <latest>1.14.0-SNAPSHOT</latest>
-    //    <release>1.13.0</release>
-    //  </versioning>
-    //</metadata>
+    // <metadata>
+    //   <versioning>
+    //     <latest>1.14.0-SNAPSHOT</latest>
+    //     <release>1.13.0</release>
+    //   </versioning>
+    // </metadata>
+    //
+    // or
+    //
+    // <metadata>
+    //   <groupId>javax.inject</groupId>
+    //   <artifactId>javax.inject</artifactId>
+    //   <version>1</version>
+    //   <versioning>
+    //     <versions>
+    //       <version>1</version>
+    //     </versions>
+    //     <lastUpdated>20100720032040</lastUpdated>
+    //   </versioning>
+    // </metadata>
     Element metadataElement = document.getDocumentElement();
     Element versioningElement = getFirstChildElement(metadataElement, "versioning");
-    if (versioningElement != null) {
-      // Note: we may want to add a flag to allow people to look for updates against
-      // "latest" instead of "release"
-      releaseVersion = versioningElement.getElementsByTagName("release").item(0).getTextContent();
-    } else {
+    if (versioningElement == null) {
       verboseLog(String.format("Could not find <versioning> tag for %s, returning null version", url));
+      return null;
+    }
+
+    // Note: we may want to add a flag to allow people to look for updates against
+    // "latest" instead of "release"
+    NodeList release = versioningElement.getElementsByTagName("release");
+    if (release != null && release.getLength() > 0) {
+      return release.item(0).getTextContent();
     }
-    return releaseVersion;
+
+    // No release info, default to the first version in the list.
```

Comment:
Based on some example metadata files it seems like this should default to the last version in the list instead of the first. e.g https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/maven-metadata.xml, https://repo1.maven.org/maven2/com/google/guava/guava/maven-metadata.xml, https://repo1.maven.org/maven2/org/apache/commons/commons-lang3/maven-metadata.xml

Reply:
Done. I also fixed the logic and tested it.